crates/server/src/authority/catalog.rs

```diff
@@ -777,34 +781,77 @@ async fn build_forwarded_response(
     }
 
     // Don't perform the recursive query if this is disabled...
-    let mut answers = if !request_header.recursion_desired() {
+    let (mut answers, authorities) = if !request_header.recursion_desired() {
```

Comment:
From here, it would be nice to find ways to reduce rightward drift by doing early returns and/or picking which match arm should be yielded out to the main body of the function.

Reply:
I was able to remove just a bit of the rightward drift, but I think doing more is going to require a rewrite of the function.